src/app/maps/m6/m6.component.ts

```diff
@@ -0,0 +1,455 @@
+import {
+  Component,
+  ElementRef,
+  OnInit,
+  QueryList,
+  ViewChild,
+} from '@angular/core';
+import { ReplaySubject } from 'rxjs';
+import { delay, first, switchMap } from 'rxjs/operators';
+
+import { MapService } from '../../map.service';
+import { PlayerService } from '../../player.service';
+
+import * as mapboxgl from 'mapbox-gl';
+import { M6Service } from './m6.service';
+import { ActivatedRoute } from '@angular/router';
+import * as marked from 'marked';
+
+@Component({
+  selector: 'app-m6',
+  templateUrl: './m6.component.html',
+  styleUrls: ['./m6.component.less'],
+})
+export class M6Component implements OnInit {
+  @ViewChild('mapEl', { static: true }) mapEl: ElementRef;
+  @ViewChild('lnk') lnk: QueryList<ElementRef>;
+
+  theMap: mapboxgl.Map;
+  samples = new ReplaySubject<GeoJSON.FeatureCollection>(1);
+  rabat = new ReplaySubject<GeoJSON.FeatureCollection>(1);
+  _sample = null;
+  info = false;
+  marked = marked;
+  accordion = [];
+  images = [];
+  name = '';
+  pics: any;
+  description: any;
+  Location: any;
+  Chapter: any;
+  Notes: any;
+  Shortcuts: any;
+  title: any;
+  features: any;
+
+  LAYER_NAME = 'place-label';
+  SECOND_LAYER_NAME = 'intro-label';
+  HIGHLIGHTS = 'highlighted';
+  SOURCE_NAME = 'samples';
+  DUMMY = { author: '', author_credits: '' };
+
+  constructor(
+    private api: M6Service,
+    private map: MapService,
+    private player: PlayerService,
+    private activatedRoute: ActivatedRoute
+  ) {
+    api
+      .getSamples()
+      .pipe(
+        switchMap((samples) => {
+          this.sample = samples.features[0].properties;
+          this.name = this.sample.name;
+          this.pics = this.sample.Images
+            ? [
+                ...this.sample.Images.map((image) =>
+                  image.width <= image.height
+                    ? { ...image, newClass: true }
+                    : { ...image, newClass: false }
+                ),
+              ]
+            : [];
+          this.description = this.sample.description;
+          this.Location = this.sample.Location;
+          this.Chapter = this.sample.Chapter;
+          this.Notes = this.sample.Notes;
+          this.Shortcuts = this.sample.Shortcuts;
+          this.title = this.Location + ' - ' + this.Chapter;
+          this.features = samples.features;
+
+          const links = document.links;
+
+          for (let i = 0; i < links.length; i++) {
+            if (links[i].hostname !== window.location.hostname) {
+              links[i].target = '_blank';
+            }
+          }
+
+          setTimeout(function () {
+            Array.from(document.querySelectorAll('.notes a')).forEach(function (

```

Comment:
very hacky and won't always work - must be a better way for this, no?

Reply:
it's good, I think we will use it like this for now